File: station_service/models/hardware.py
"""
Hardware status model definitions.
"""

from typing import Any, Dict, Optional

from pydantic import BaseModel


class HardwareStatus(BaseModel):
    """Hardware device status."""

    id: str  # "dmm"
    driver: str  # "KeysightDMM"
    status: str  # "connected", "disconnected", "error"
    connected: bool
    last_error: Optional[str] = None
    config: Dict[str, Any]  # Applied configuration
    info: Optional[Dict[str, Any]] = None  # Device info (IDN, etc.)

File: station_ui/src/types/hardware.ts
/**
 * Hardware type definitions.
 */

/**
 * Hardware connection status.
 */
export type HardwareConnectionStatus = 'connected' | 'disconnected' | 'error';

/**
 * Hardware device status.
 */
export interface HardwareStatus {
  /** Hardware ID (e.g., "dmm") */
  id: string;
  /** Driver name (e.g., "KeysightDMM") */
  driver: string;
  /** Connection status */
  status: HardwareConnectionStatus;
  /** Whether connected */
  connected: boolean;
  /** Last error message */
  lastError?: string;
  /** Applied configuration */
  config: Record<string, unknown>;
  /** Device info (IDN, etc.) */
  info?: Record<string, unknown>;
}
